src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
Here's the code as David originally wrote it, assuming we had a big Json::Value behind it:

    Json::Value& obj = txns.append (Json::objectValue);
    obj[jss::tx_blob] = strHex (item->peekData ());
 
So you can see the same pattern - the difference is only that append is a method on `Json::Value` but has to be implemented as a free function in the generic code...

That said, I have a semantic sugar change to make it seem even more like the original, moment please...